package.json

```diff
@@ -27,7 +27,7 @@
   },
   "devDependencies": {
     "@jest/globals": "^29.3.1",
-    "@types/jest": "^29.2.5",
+    "@types/jest": "^29.5.0",
```

Comment:
See my comment in the PR summary. There were some Jest internals that weren't being properly exposed, I think it's somewhat safe to update our typing package (to be honest I'm not sure if this package version is specifically aligned with Jest).